dvc/external_repo.py

```diff
@@ -45,117 +36,49 @@ def __init__(self, url, rev, cause):
         )
 
 
-class ExternalRepo(object):
-    REPOS_DIR = "repos"
-
-    PARAM_URL = "url"
-    PARAM_VERSION = "rev"
-
-    SCHEMA = {Optional(PARAM_URL): str, Optional(PARAM_VERSION): str}
+def _install(url=None, rev=None, rev_lock=None, cache_dir=None):
+    import git
+    from dvc.repo import Repo
 
-    def __init__(self, dvc_dir, **kwargs):
-        self.repos_dir = os.path.join(dvc_dir, self.REPOS_DIR)
-        self.url = kwargs[self.PARAM_URL]
+    _path = tempfile.mkdtemp("dvc-repo")
 
-        self.name = "{}-{}".format(os.path.basename(self.url), hash(self.url))
+    try:
+        repo = git.Repo.clone_from(url, _path, depth=1, no_single_branch=True)
+    except git.exc.GitCommandError as exc:
+        raise InstallError(url, _path, exc)
 
-        self.rev = kwargs.get(self.PARAM_VERSION)
-        self.path = os.path.join(self.repos_dir, self.name)
+    try:
+        revision = rev_lock or rev
+        if revision:
+            try:
+                repo.git.checkout(revision)
```

Comment:
does it checkout the revision if it's not fetched yet? (since we use depth=1)

Reply:
Great point!